actions/common/search_models.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/datastore"
+)
+
+// ConditionsModel is a generic model for handling conditions with metadata
+type ConditionsModel[TFilter any] struct {
+	// Custom conditions used for filtering the search results
+	Conditions TFilter `json:"conditions"`
+	// Accepts a JSON object for embedding custom metadata, enabling arbitrary additional information to be associated with the resource
+	Metadata *engine.Metadata `json:"metadata,omitempty" swaggertype:"object,string" example:"key:value,key2:value2"`
+}
+
+// SearchModel is a generic model for handling searching with filters and metadata
+type SearchModel[TFilter any] struct {
+	ConditionsModel[TFilter]
+
+	// Pagination and sorting options to streamline data exploration and analysis
+	QueryParams *datastore.QueryParams `json:"params,omitempty" swaggertype:"object,string" example:"page:1,page_size:10,order_by_field:created_at,order_by_direction:desc"`
```

Comment:
Is this really possible to use those as query params?
Beside this, maybe it's good moment to name it differently, for example `Paging` or `Pagination` as we will need to make changes in clients.

Reply:
It seems like out of scope of this task - but we can discuss it.

Reply:
Ok, lets do that in separated task.